### DISCUSSION TITLE
Multiple annotations

### DISCUSSION THREAD

**Discussion by IceDragon200** [kdl-org/kdl#245] [Ideas]
So annotations, the primary README briefly touches on it.

What's interesting, originally it seems to have been used for type annotating:

```
(date)"2021-10-24"
```

But recently reading the data model https://github.com/kdl-org/kdl/issues/225, it now goes by the name 'tag' which suggests it is now a general purpose value.

So would it make sense to allow multiple tags or annotations?

```kdl
(AI)(Enemy)entity {
  name "Box"
  position 0 0
}

(AI2)(Enemy)entity {
  name "Goblin"
  position 0 1
}

(Controllable)(Player)entity {
  name "Player"
  position 3 3
}
```

Not the perfect example, but I think it gets the point across.

**Comment by larsgw** [comment]
For values it is still used for type annotating, for entities it is about indicating the relationship between a node and its parent node (or the root document). In your example, I think most of the tags/annotations could also be properties (`Controllable` at least).

**Comment by dezren39** [comment]
this would be nice because properties may be being used for other purposes, especially in describing complex graph data or namespaces. while you could put these as properties or values, by letting this kind of type/tag info be annotated on top opens up a lot of capabilities downstream for users who might be consuming a format using kdl. i'd prefer if it stored the annotations in an array then some apps could use the ordering for defining essentially inheritance, while other apps could use the information as unordered tags or disregard it altogether.

i'd even like if we could add annotations to properties too, in addition to values and nodes. but that's another thing.
